### ISSUE TITLE
Local variable type hints. Helps identify local variable return types

### ISSUE THREAD

**Issue by TPGamesNL** [SkriptLang/Skript#3864]
Local variable type hints were implemented in 2.2-dev37c, see [dev37c release notes](https://github.com/SkriptLang/Skript/releases/tag/dev37c) and [bug report](https://github.com/SkriptLang/Skript/issues/1379). These were accidentally disabled a couple months later (I think it was accidentally, but I'm not sure tho, GitHub isn't on my side for this one).

**The question is: should these be brought back?**
The implementation will give a warning when loading the script, when the determined type of a simple local variable (a simple local variable is one like `{_a}`, `{_item}`, `{_l::1}`, and not `{_l::%{_i}%}` or `{notLocalVariable}`) doesn't match the required type for its usage. For example, this code would produce such a warning:
```
set {_x} to "Hello, world!"
set {_y} to location of {_x}
```
since `{_x}` is a string, which doesn't have a location.

Another thing to keep in mind is addons, since some addons may change variables in different ways, this is often done when an effect is ran asynchronously, but it has a result that can be stored in a variable, for example:
- [skript-db's execute statement](https://github.com/btk5h/skript-db#effect-execute-statement)
- [skript-mirror/skript-reflect custom syntax (with parse section)](https://tpgamesnl.gitbook.io/skript-reflect/advanced/custom-syntax), because local variables are carried over from the parse section.

These addons would have to be updated, otherwise using them with local variables will produce the warnings.

Any feedback is appreciated (:

**Comment by TheDGOfficial** [comment]
Yes, they are only warnings and should be probably added back. I remember it being working before and actually helped me catch some things. With a config option to disable them of course, like most other warnings.

Not related to this, but GitHub has a new discussions feature and a new issue system. Maybe look at it and enable for this repo?

As for add-ons, they should be updated if the problem can only be fixed on their side. Otherwise, AsyncEffect or other things maybe updated to correct these, if possible.

**Comment by TPGamesNL** [comment]
I've actually just thought of something that might make it so that addons don't need updating: updating the type hints when `Variable#acceptChange` is called. While this isn't exactly something that should be done on `acceptChange`, it would prevent issues with addons, so long as the addons use the expression changers, and not `Variables.setVariable`.

**Comment by TPGamesNL** [comment]
> I've actually just thought of something that might make it so that addons don't need updating: updating the type hints when `Variable#acceptChange` is called. While this isn't exactly something that should be done on `acceptChange`, it would prevent issues with addons, so long as the addons use the expression changers, and not `Variables.setVariable`.

A lot of addons unfortunately just `expr instanceof Variable` instead of checking `acceptChange`, so this won't help

**Comment by TheLimeGlass** [comment]
Waiting on pull https://github.com/SkriptLang/Skript/pull/4566 before starting this.
